bpftools/gen_tcpsyn_data.py

```diff
@@ -0,0 +1,68 @@
+import argparse
+import string
+import sys
+
+def gen(params, l3_off=0, ipversion=4, negate=False):
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        prog="%s tcpsyn_data --" % (sys.argv[0]),
+        description=r'''
+
+Generate raw BPF rules matching SYN packets with a payload.
+It works on both IPV6 and IPV4. IPV6 extension headers are not supported.
+This rule returns match if the packet is SYN only and has a IP total length field
+which is bigger than IP header length added to the TCP header length.
+
+If this bigger that means there are some data -> this matches SYN packet with data.
+It does not match all the other packets.
+
+IPV4 successfully manually tested on 25 packets.
```

Comment:
No need for that sentence in help.

please document if the flags are checked. If only SYN must be set for a positive match, or at most SYN must match. ie: RST+SYN with payload will match? or SYN+ACK?